neqo-transport/src/connection/mod.rs

```diff
@@ -1945,6 +1941,12 @@ impl Connection {
             return Ok(());
         }
 
+        // Check if there is a Datagram to be written
+	// Currently we're giving them priority over user streams; they could be moved
+	// to after them (after Normal)
+        self.quic_datagrams
+            .write_frames(builder, tokens, &mut self.stats.borrow_mut());
+
```

Comment:
I think that you want two changes here.

1. use `stats` in place of `&mut self.stats.borrow_mut()` as this is already borrowed (I'm somewhat surprised that this compiled)
2. Check `if builder.is_full()`.

Are there no tests?  Maybe there should be...  (Sorry to do that to you.)